From 062c9d425d39b81c5cbeaf03974e2a5b2ef9b766 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 18 Aug 2022 15:02:24 +0300
Subject: [PATCH] [SW-86869] gaudi3: remove security en check in rom recovery

Since ROM recovery is not called from erom recovery flow in case
of erom manifest load and verification failure, this patch adds
changes to remove security_enable fuse check for bringing up
PCIe, I2C link and disable range register. ROM recovery will only
be called on non-secured system and doesn't need security_enable
fuse check.

Change-Id: I836ed59bb887466965c3fae02e71ac4baa5d2541
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 429cab7..c544974 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -188,18 +188,15 @@
 
 static void execute_rom_recovery(void)
 {
-	int sec_en = security_efuse_security_en();
 	arcboot_rom_status_update(PROGRESS_ARCBOOT_IN_ROM_RECOVERY);
 
-	if (!sec_en) {
-		security_rr_recovery_config();
-		rom_pcie_recovery();
-	}
+	security_rr_recovery_config();
+	rom_pcie_recovery();
 
 	rom_d2d_link_init();
 
-	if ((!sec_en) && ((arcboot_rom_global.config_fuse &
-		EFUSE_SYSCFG_DISABLE_TWI_IF_MASK) == 0))
+	if ((arcboot_rom_global.config_fuse &
+		EFUSE_SYSCFG_DISABLE_TWI_IF_MASK) == 0)
 		security_i2c_s_enable();
 
 	arcboot_rom_status_update(PROGRESS_ARCBOOT_ROM_RECOVERY_END);
